### PR TITLE
common/options: validate see-also

### PR DESCRIPTION
**Pull request by tchaikov**
Fixes: https://tracker.ceph.com/issues/51483

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by ideepika — APPROVED** [review]
tested locally, lgtm!

**Comment by ideepika** [comment]
shaman failure looks related, 
```
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/arm64/AVAILABLE_ARCH/arm64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7149-g46305a89/rpm/el8/BUILD/ceph-17.0.0-7149-g46305a89/src/common/options/y2c.py", line 366, in <module>
    main()
  File "/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/arm64/AVAILABLE_ARCH/arm64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7149-g46305a89/rpm/el8/BUILD/ceph-17.0.0-7149-g46305a89/src/common/options/y2c.py", line 360, in main
    translate(opts)
  File "/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/arm64/AVAILABLE_ARCH/arm64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7149-g46305a89/rpm/el8/BUILD/ceph-17.0.0-7149-g46305a89/src/common/options/y2c.py", line 249, in translate
    with open(opts.input) as infile, \
FileNotFoundError: [Errno 2] No such file or directory: '/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/arm64/AVAILABLE_ARCH/arm64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7149-g46305a89/rpm/el8/BUILD/ceph-17.0.0-7149-g46305a89/aarch64-redhat-linux-gnu/src/common/options/mds-client.yaml'
````

**Review by ideepika — CHANGES_REQUESTED** [review]
shaman failure looks related,

```
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/arm64/AVAILABLE_ARCH/arm64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7149-g46305a89/rpm/el8/BUILD/ceph-17.0.0-7149-g46305a89/src/common/options/y2c.py", line 366, in <module>
    main()
  File "/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/arm64/AVAILABLE_ARCH/arm64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7149-g46305a89/rpm/el8/BUILD/ceph-17.0.0-7149-g46305a89/src/common/options/y2c.py", line 360, in main
    translate(opts)
  File "/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/arm64/AVAILABLE_ARCH/arm64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7149-g46305a89/rpm/el8/BUILD/ceph-17.0.0-7149-g46305a89/src/common/options/y2c.py", line 249, in translate
    with open(opts.input) as infile, \
FileNotFoundError: [Errno 2] No such file or directory: '/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/arm64/AVAILABLE_ARCH/arm64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-7149-g46305a89/rpm/el8/BUILD/ceph-17.0.0-7149-g46305a89/aarch64-redhat-linux-gnu/src/common/options/mds-client.yaml'
```

https://shaman.ceph.com/builds/ceph/wip-deepika-testing-2021-08-23-1343/

**Comment by tchaikov** [comment]
no, the build failure is not related
```
make[2]: *** Deleting file 'src/common/options/mds-client.yaml'
[  1%] Generating mds-client.yaml
CMake Error at configure_file.cmake:1 (configure_file):
  configure_file Problem configuring file


make[2]: *** 
```

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-kefu-testing2-2021-08-23-2127/fe751cc12c7a2ff35d4f8fd6db46c22155f00bf3/

**Comment by tchaikov** [comment]
sorry, i was wrong. it is related. but i still don't get why.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-51483-kefu/aa1636a14e0f6a215afad05ad90340b52c281f32/

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-51483-3-kefu/c08bcc1cd13be5631bd6a20327bcc8b3ce9ec266/

File: src/common/options/CMakeLists.txt [hunk 1]
```diff
@@ -100,5 +103,6 @@ add_options(rgw)
 
 add_library(common-options-objs OBJECT
   ${common_options_srcs})
-add_custom_target(legacy-option-headers
-  DEPENDS ${legacy_options_headers})
+
+add_ceph_test(validate-options
```
Comment (ideepika) [inline src/common/options/CMakeLists.txt:107]:
```suggestion
include(AddCephTest)
add_ceph_test(validate-options
```
@tchaikov

File: src/common/options/CMakeLists.txt [hunk 2]
```diff
@@ -100,5 +103,7 @@ add_options(rgw)
 
 add_library(common-options-objs OBJECT
   ${common_options_srcs})
-add_custom_target(legacy-option-headers
```
Comment (tchaikov) [inline src/common/options/CMakeLists.txt:103]: Remove this by accident. Will fix it tomorrow.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-51483-5-kefu/f62fbb02780dde4d22de89ee68c8451d712cc52c/

**Comment by tchaikov** [comment]
@ideepika could you take another look?

**Comment by ideepika** [comment]
@tchaikov make check failure is related, can you fix it:
```
  3/243 Test   #1: validate-options ..........................***Failed    0.11 sec
[Errno 2] No such file or directory: ' /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/global.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/cephfs-mirror.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/crimson.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/mgr.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/mds.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/mds-client.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/mon.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/osd.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/rbd.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/rbd-mirror.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/immutable-object-cache.yaml /home/jenkins-build/build/workspace/ceph-pull-requests/build/src/common/options/rgw.yaml'

        Start  37: unittest_base64
```

**Review by ideepika — APPROVED** [review]
lgtm!

**Pull request merged**